### ISSUE TITLE
Home Assistant envisalink testing help wanted!

### ISSUE THREAD

**Issue by ufodone** [ufodone/envisalink_new#1]
Hi all!  Sorry in advance for the unsolicited message.  

I'm in the process of modernizing the Home Assistant `envisalink` integration.  This is a fairly substantial change and I'd like to get as much testing on it as possible before submitting it as a PR back to the HA.  

As you probably know, the integration supports both DSC and  Honeywell systems.  While much of the code is the same, there are differences and I have no ability to test changes to the Honeywell portion as I only have a DSC system myself.  The integration also supports EVL3 and EVL4 units and I only have an EVL4.

And because the changes include refactoring how the configuration of the system is done (moving from configuration.yaml into the UI), this also means that variances in configuration may expose bugs that I don't see in my own testing.

To this end, I'm reaching out to you all in the hopes that some will be willing to help test the changes.  I wasn't sure the best was to communicate with people using this integration so I went through many of the issues that have been opened for it in the Home Assistant project.

The lack of support for UI configuration has in some ways hampered my ability to make other updates to the integration (such as reintroduction of the zone bypass switches) so I'm hopeful that if I can get these changes contributed back I'll be in a better position to support the integration.

I'm setting this up as a HACS custom component for now using a different name.  This should allow easier testing and iteration while will making it quick and easy to switch back to the standard/build-in integration.  While it's not quite ready for external use yet, I hope to have a version ready very soon.

I would appreciate any help people can offer - be that testing the integration, providing debug logs from Honeywell systems (so maybe I can make an emulator to test against) or even just posting your existing configurations and details on your setups (automations, etc.) to help me try and cover all the bases.

[Here](https://github.com/ufodone/envisalink_new#readme) is a list of the major changes to date.

If you think you can  help in any way, please let me know below.  Thanks in advance!

@ChirpyTurnip @Damien2s @DanielBaulig @Hamish1066 @Papester @TheFuzz4 @abarbaccia @abmagfab @active-atom
@adman120 @audymatic @ausfas @bbrendon @bolapara @camrun91 @cgraf @crackers8199 @facuzestoa @greggitter
@jl-678 @larry-glz @mniquette @monxhy @nate3421 @olddawgpowers @raspman @rct @ryanpie86 @schmuy @tinuva
@undert03 @warnejo

**Comment by schmuy** [comment]
Happy to give it a go.  Config just as it was before but with different name?

**Comment by jl-678** [comment]
I am happy to try it too.

**Comment by bolapara** [comment]
I have a dual DSC setup I can test on.

**Comment by camrun91** [comment]
I am more than happy to help however I can. I have a DSC system

**Comment by ufodone** [comment]
Thanks for the offers!  I've got a bit more cleanup to do first and then I'll add some instructions on how to get it setup as well as things I'm interested in tested on different configs.  I hope to try and get things ready to go sometime tonight.

**Comment by cgraf** [comment]
Hey ufodone. I'm happy to help out if I can. I think I have a Vista 20P
(ADT Safewatch 3000). Ancient system. I am tied into Envisalink monitoring,
so hopefully we won't be doing a lot of things that might summon the
police.

**Comment by ChirpyTurnip** [comment]
Happy to help too! DSC + EVL4....  :-)

**Comment by ausfas** [comment]
I have Honeywell vista 21p mad happy to test it

**Comment by ufodone** [comment]
Hi all, thanks again for offering up your help!

I think the integration is ready for some testing now.  It seems to be working well with my own system.  Release v0.1.1 should be available once you install the integration via HACS.

I've added some more info to the README describing how the new configuration system works and listed some things I'm interested in getting feedback on.  But generally speaking, let me know if anything looks out of place, etc.

If you find something not working or have any other feedback, please open a separate issue for it so this one doesn't get too large.

**Comment by ufodone** [comment]
Oh, and it's worth nothing that I am speculating that the EVL3 and EVL4 have almost identical interfaces based on what's in the code and the documentation I've found.  But if you have an EVL3, I might suggest you trying hitting the following links on your browser before trying this integration out:

`http://envisalink_ip/2` -> This one should return a page which somewhere has `Security Subsystem - DSC` (or Honeywell)

`http://envisalink_ip/3` -> This one should return a page which lists the firmware version and MAC at the bottom.

(Substitute `envisalink_ip` for the IP address or hostname of your EVL.)
(You'll first be prompted for your username and password which will be the same as you use in HA.)

If these pages don't work or return that data, please let me know and hold off on trying this integration out.

**Comment by jl-678** [comment]
One question, do we need to disable or uninstall the current integration to test out this version? I want to be sure that I am following the correct process.

**Comment by greggitter** [comment]
Just in case, the testing information and documentation is posted here:
https://github.com/ufodone/envisalink_new

**Comment by greggitter** [comment]
@ufodone Maybe I'm missing something but I am not finding envisalink_new in hacs, nor in settings | add integration.  I assumed the first step was to rename the configuration to envisalink_new but that results in a HA configuration error.

**Comment by greggitter** [comment]
Ah...thanks!

**Comment by jl-678** [comment]
Do you do anything with the current integration when you install this one?

**Comment by greggitter** [comment]
I did not change my current integration (didn't rename yet), added the new, entered the host and credentials using the same name "Home Alarm" and it's stuck on initializing.  Maybe I created a conflict.  I can move around in HA but returning to the integrations page, it still indicates initializing.  I can click to configure it and the config window pops up.

**Comment by ufodone** [comment]
@greggitter I've opened an issue for this: #3

**Comment by ufodone** [comment]
FYI that v0.2.1 and v0.2.2 are broken for Honeywell systems so please do not upgrade to them if you have a Honeywell panel.  I'll get an update out tonight.

**Comment by ChirpyTurnip** [comment]
Sorry for the delay, had to wait for weekend!

*Implementation steps for DSC + EVL4:*

1. Changed envisalink: section of configuration.yaml to envisalink_new:
2. Added custom HACS repo, installed envisalink_new integration
3. Restarted HA
4. New integration detected automatically, configuration correctly imported
5. Commented out envisalink_new: section of configuration.yaml
6. Restarted HA
7. Verified settings (e.g. zone dump interval), sensor check, arm/disarm
check, dashboard and Node Red linkages
8. Everything looks good!
9. Remove everything, re-add it all manually
10. Repeat tests. All OK.

Does it import your configuration.yaml correctly? - **YES**
If you don't have any configuration.yaml entries, are you able to successfully
add the EVL via the UI? - **YES**
Does it auto-detect your EVL version and panel type correctly? - **YES**
Do you see your firmware version listed in the device settings? - **YES**
Do you see any errors or warnings in the log files? - **NO** (log are clean
for custom_components.envisalink_new)
If you have any automations, etc. connected to your envisalink, do they
still work correctly? - **YES**
Generally speaking, does anything look wrong or broken? **NO** (everything
seems to be good, will leave this in place for now).

Excellent work!

:-)


On Sat, 14 Jan 2023 at 07:55, ufodone ***@***.***> wrote:
>
> FYI that v0.2.1 and v0.2.2 are broken for Honeywell systems so please do
not upgrade to them if you have a Honeywell panel. I'll get an update out
tonight.

**Comment by ufodone** [comment]
I've release [v0.3.0](https://github.com/ufodone/envisalink_new/releases/tag/v0.3.0) now with updates for Honeywell systems.  The changes are fairly substantial and I had limited ability to test this myself as I don't have a Honeywell system.

I'm in the process of building a mock EVL server to test against but I'm primarily going on the (dated) documentation I have about the EVL.  I'd really appreciate it if anyone with a Honeywell system could provide a debug log of their system showing zones opening and closing as well as arming and disarming in the various modes.  That would give me great info to go on to improve the mock server I have so I can test future changes better.

The logs could be either from this release or from a stock HA `envisalink` integration.  

```
logger:
  default: info
  logs:
    homeassistant.components.envisalink: debug
    pyenvisalink: debug
    custom_components.envisalink_new: debug
```

As always, you may want to scrub your password and alarm code from the logs.  As I'm now a bit more familiar with the Honeywell portion of the code, I realize the scrubbing the alarm code is trickier given how it is sent to the EVL so any arm/disarm actions you take would be best done from a physical keypad so no code shows up in the logs.

Thanks for any help you can provide!

**Comment by ryanpie86** [comment]
Sorry I missed this. Count me in.

**Comment by facuzestoa** [comment]
HiGlad to see improvements coming!I’m running EVL4 on DSC. I will install and test as soon back home. Don’t like to do it on remote 😉Saludos,FacundoOn 14 Jan 2023, at 15:53, ryanpie86 ***@***.***> wrote:﻿
Sorry I missed this. Count me in.

**Comment by ufodone** [comment]
@facuzestoa @ryanpie86 Thanks for the offer.

@greggitter has been giving me a ton of help with the Honeywell side of things as I don't have one myself.  I think it's in decent shape now with the [v0.3.4](https://github.com/ufodone/envisalink_new/releases/tag/v0.3.4) release.

**Comment by larry-glz** [comment]
I just tested on my honeywell:

Does it import your configuration.yaml correctly?  **Yes**
If you don't have any configuration.yaml entries, are you able to sucessfully add the EVL via the UI?  **Didn't test.**  
Does it auto-detect your EVL version and panel type correctly?  **Yes**
Do you see your firmware version listed in the device settings?  **Yes**
Do you see any errors or warnings in the log files?  **No**
If you have any automations, etc. connected to your envisalink, do they still work correctly?  **Yes (so far)**
Generally speaking, does anything look wrong or broken?  I'll report if i see anything.  I plan to leave this version enabled.

**Comment by rct** [comment]
@ufodone - Sorry I missed this until today when @tinuva pointed it out. 

I will give it a try and let you know. Though like @tinuva I intend to run it connecting to my existing AlarmServer that I use as a proxy to my EnvisaLink

Note: I [ran into a problem with new Envisalink EVL-4 firmware](https://github.com/juggie/AlarmServer/issues/78), which prompted me to dig into things.  I created [this current state of AlarmServer discussion on Juggie's AlarmServer repo](https://github.com/juggie/AlarmServer/discussions/80) https://github.com/juggie/AlarmServer/discussions/80

**Comment by ufodone** [comment]
Hi all... a couple quick updates:

Thanks again to everyone who's been helping me test these changes out.  Your input and feedback has been invaluable.

The v0.4.0 release had an unfortunate bug that prevented zone and partition state changes from showing up in Home Assistant.  I've published a new version which fixes that issue.  I didn't notice it originally because I'd been focusing on some other areas.  Sorry!

The latest [v0.4.1](https://github.com/ufodone/envisalink_new/releases/tag/v0.4.1) release has some changes to how zone timer dumps are processed.  I've seen plenty of issues opened related to this code over the past couple of years and so took a stab and making it work more reliably based on what I've learned about how the DSC and Honewell system report their data.  I hope the changes will make things more reliable and I'd appreciate you keeping an eye open for any oddities. 

Please also note that for some releases it may be necessary to remove the integration completely via HACS and then re-install it.  If you don't, you may see odd behaviour (such as duplicate devices).  This is due to me needing to change some of the details on how the device and entities are registered.  I think that stuff is fairly stable now so hopefully this won't be a regular occurrence.

**Comment by schmuy** [comment]
I was just about to comment that zones weren't updating in 0.4.0.

Thanks for all your work on getting this done and having updates so quickly.

**Comment by ChirpyTurnip** [comment]
Hi,

I tried to update to the new version (DSV/EVL4) and HA wouldn't complete
the restart. I removed the integration and re-added and it refuses to
connect to the EVL. I've rolled back to v3.5, and it connected immediately.
Unfortunately all my zone names etc are missing so the entities names are
different now from the looks of it so I will need to unpick a few things....

Any tests you want me to run?

Thanks!

**Comment by ufodone** [comment]
@ChirpyTurnip Let's take it over [here](https://github.com/ufodone/envisalink_new/issues/12) to avoid spamming others.

**Comment by madsci1016** [comment]
Are testers still needed? Just found this while debugging the random zone open/close issue with the old integration. Would love to see a more polished version that is more stable.

**Comment by ufodone** [comment]
@madsci1016 Absolutely!  What started out as just trying to get the integration using the config flow has turned it to a bit of a "fix everything" project so happy to have more folks testing and providing input.

**Comment by garyak** [comment]
Single EV4, Honeywell Vista 20-P, yaml configuration on HAOS 9.5. I'd like to help with testing. I prefer "manual" installation of critical components. I've downloaded the latest release(v0.4.1 into config/custom_components. What steps are needed to swap envisalink for envisalink_new?

**Comment by greggitter** [comment]
> What steps are needed to swap envisalink for envisalink_new?

Here:  https://github.com/ufodone/envisalink_new

**Comment by ufodone** [comment]
@garyak 

>I prefer "manual" installation of critical components.

I don't have any plans at the moment to remove the configuration.yaml method.  However, any new configuration items will only be available via the UI.  This is a Home Assistant [requirement](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md) so not anything I have control over.

**Comment by garyak** [comment]
Thanks. I started testing. I have other components that I originally installed manually. Later, the dev moved the component to an "official" integration. I was able to add the new integration without using HACS. is this an optional path?

envisalink_new on HA 2023.2.3 picked up the old envisalink config from configuration.yaml and connected successfully. If I subsequently comment out the old config and restart, envisalink_new doesn't start. For "manual" installations, I need to retain the yaml config, correct? I'm still using the envisalink: header in yaml.

**Comment by ufodone** [comment]
Yes, you can install the integration manually rather than via HACS if you prefer.

> If I subsequently comment out the old config and restart, envisalink_new doesn't start

You need to rename "envisalink:" to "envialink_new:" in your configuration.yaml and the new integration will pick it up.  The name needs to be different to avoid a conflict with the stock integration.  This is temporary for now but will eventually be renamed back to envisalink when I submit the changes back into the main HA project.

> I need to retain the yaml config, correct? I'm still using the envisalink: header in yaml.

That's up to you but probably recommended.  Once you've done the rename, the new integration should just pick it up and you'll see your EVL in the Settings->Devices&Services screen and shouldn't need to do anything else.

The alternative would be to comment out all your config and then manually add the device via the UI.  This path is likely for new users of the integration who don't have any existing config.

**Comment by garyak** [comment]
Integration isn't picking up the yaml as envisalink_new. Renaming back to envisalink allows the integration to read the config on startup.  Here's the integration shown on the Devices&Services screen:

![envisalink](https://user-images.githubusercontent.com/47036329/217889850-53ac4d8a-dc2c-4397-bd0c-92e86c7e1f01.jpg)

Needed to comment out the envisalink config and restart to allow me to add the envisalink_new integration. Once added, the yaml config needs to be in place for HA to load envisalink_new at startup. I haven't found the config entity.

**Comment by ufodone** [comment]
Let's take it up over [here](https://github.com/ufodone/envisalink_new/issues/16).

**Comment by rct** [comment]
Re: manual YAML config, `envisalink` -> `envisalink_new` - I don't know whether to add this here, or to the closed #16.   

Is this expected to work, or is the GUI config flow required at this point?

Trying to test the new integration using manual YAML configuration for the envisalink and all zones.  

1. Config key renamed from  `envisalink` vs. `envisalink_new`.
2. Restarting Hass, it doesn't find the config keys for `envisalink`

```

2023-02-11 15:11:04.161 ERROR (MainThread) [homeassistant.config] Invalid config for [envisalink]: required key not provided @ data['envisalink']['host']. Got None
required key not provided @ data['envisalink']['panel_type']. Got None
required key not provided @ data['envisalink']['password']. Got None
required key not provided @ data['envisalink']['user_name']. Got None. (See ?, line ?). Please check the docs at https://www.home-assistant.io/integrations/envisalink
2023-02-11 15:11:04.168 ERROR (MainThread) [homeassistant.setup] Setup failed for envisalink: Invalid config.
2023-02-11 15:11:07.718 ERROR (MainThread) [custom_components.envisalink_new.pyenvisalink.alarm_panel] Unable to validate connection: Cannot connect to host envisalinkhost:8080 ssl:default [Connect call failed ('192.168.xxx.yyy', 8080)]
```

In the Hass GUI `.../config/integrations` it doesn't show a card for Envisalink or Envisalink_new, but when I click *+ Add integration*, it shows 

![image](https://user-images.githubusercontent.com/38770/218279502-cbf36f6c-4c9d-4bd8-ae76-ab03a7f9e002.png)


**Note:**  Judging by that last line in the error message,  It did partially pickup the configuration for envisalink host, because it tried to connect to the envisalinkhost, but on port 8080, and possibly trying to do SSL?

`Unable to validate connection: Cannot connect to host envisalinkhost:8080 ssl:default [Connect call failed ('192.168.xxx.yyy', 8080)]` 

So at this point, do I need to configure  host, panel_type, user_name, password through the add integration GUI/config flow?

**Comment by ufodone** [comment]
FYI I have published a new release with some fairly substantial changes so please read the release notes carefully before upgrading.   All the details are in the release notes [here](https://github.com/ufodone/envisalink_new/releases/tag/v0.5.0).

One other thing I forgot to add to the release notes:

I've changed to port used to discover the EVL's MAC address back to port 80 (it was previously 8080).  If you firewall or port-forward to your EVL, you may need to adjust that to account for the new port.

Thanks!

**Comment by garyak** [comment]
This is likely a local network issue, but thought I'd mention it for error handling.
```
2023-02-11 21:07:02.925 ERROR (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Command '00' failed due to timeout waiting for response from EVL
2023-02-11 21:07:02.933 ERROR (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Caught unexpected exception: IncompleteReadError('0 bytes read on a total of undefined expected bytes')
2023-02-11 21:07:02.942 ERROR (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Reconnection attempt in 30s
```
Running latest release, Only one event like this overnight. Reconnection was successful.

**Comment by rct** [comment]
0.5.0 testing:  DSC/EVL4/01.05.204 seems to be working in general. Tested disconnecting/reconnecting by killing my AlarmServer proxy, uncaught exception reported in #20.

**Comment by ufodone** [comment]
Hi all! I've published a new v0.6.2 release which contains a new algorithm for handling state changes on Honeywell panels.  If you have a Honeywell system, please keep an eye out for any oddities.

Also, please ensure that you disable zone timer dumps: `zonedump_interval: 0` in your configuration.yaml or find the setting in Configuration->Advanced.  If you don't disable them, it will likely interfere with the new algorithm.

If you're interested in the details, you can find some in #9.

If you find an issue, please open a new issue to track it.

Thanks to @philh30 for the changes.

The zone timer dumps were mainly for Honeywell panels in the past.  With this new algorithm, I think they may be completely unnecessary.  If you have a DSC panel and feel they are still required, please let me know.

**Comment by larry-glz** [comment]
I’ve seen some weird behavior with the recent update, but I can’t seem to replicate. It appears (not sure) that during arming, all the zones are tripped. But the system is still able to be armed. So the Envisalink seems ok, it may be the integration logging the trips. When this happens and I disarm, all is cleared. I’ll keep monitoring.

**Comment by ufodone** [comment]
I think things are in pretty good shape now to start the process of getting these changes merged back into Home Assistant.

Please let me know if there are any issues are you still encountering. 

Thanks!

**Comment by larry-glz** [comment]
Looks solid for me (Honeywell, EV4).  Good luck and thanks for the effort!

**Comment by ufodone** [comment]
A bit of unfortunate news.  I tried to submit the changes back to HA but they won't accept them unless I break the changes into a series of smaller PRs for the various different things that have changed.  

This project has already taken significantly more time than I'd originally intended and I don't have the additional time right now to re-organize all the changes like they've asked.  So for the foreseeable future, I'm going to leave these changes here as a custom component.

I will continue to support and develop against this version.

Thank you all for helping test these changes and providing the valuable feedback.  It has been immensely helpful and I appreciate the time you've taken.

**Comment by jl-678** [comment]
I am sorry to hear this especially after your extensive efforts. I agree that leaving it as a custom component makes sense given the circumstances. I wonder if anyone else on this thread or a future visitor might be able to help with this.

**Comment by garyak** [comment]
First, let me thank you for your work to simplify and integrate Envisalink systems into HA. The Envisalink integration is what first attracted me to HA several years ago. I'll use your work as a custom component until it replaces the old HA integration sometime in the future. 

Also know that you're not the first developer I've assisted with testing to encounter the piece-meal PR obstacle. As purely a consumer, not a developer, I can only guess the development process of multiple, independently tested changes is for code stability and inspection purposes.  Perhaps greater visibility of this requirement in the developer docs would reduce developer rework in the future. 

Thanks again and I look forward to your future development on this or different component.

**Comment by ufodone** [comment]
I definitely understand and respect their position.  I hope my comments did not leave a different impression. 

But this work has consumed a good chunk of my free time for the past few months and I honestly just don't have the energy right now to go through this process.  Hopefully I'll find that energy in the future.  Sorry.

**Comment by larry-glz** [comment]
Thank you for all the time and effort you've dedicated to this project.  Many of us truly appreciate.  While it's unfortunate you need to piece-meal the update to the core integration, having the project remain active as a custom component is just as good.  Thanks again!

**Comment by sheyman1** [comment]
So sorry to hear that.  Agree that it is unfortunate.  Thank you for all of
your hard work on this project, you have helped a lot of people.

Scott

**Comment by rct** [comment]
@ufodone - Thank you for all the time you have put in on this and essentially picking up a somewhat abandoned project.

Brief bit of opinion:

> I think I somewhat understand both positions here.  The code base for Home Assistant has become very large with an enormous number of contributors.  All that contributed code runs inside of their project.  So I can understand having a high bar for accepting code changes.
> 
> However, there are many integrations that don't have a lot of developer resources or a large user base that need a way of decoupling smaller integrations and lowering the barrier for people like yourself to continue doing the right thing.  The existing [envisalink integration has 739 users](https://www.home-assistant.io/integrations/envisalink) (that have opted into sharing usage data). That's [0.4% of Home Assistant installations according to their stats](https://analytics.home-assistant.io/integrations/).  Still the envisalink integration ranks 243 out of 986 integrations. 
> 
> That being said, I don't think anything is going to change soon.

So, what are the next steps and how can we help?

Should people be encouraged to install envisalink_new via HACS from this repo?

Will you publish your updated pyenvisalink to pypi?

Do you want contributions for updated README's for these two repos?

**Comment by ufodone** [comment]
> So, what are the next steps and how can we help?

TBH I don't know.  The most realistic next step is probably for me to get the energy to break the changes apart into separate PRs.  I'm just not prepared for that time commitment right now (and am honestly still reeling a bit from the feedback from HA that this will be required).  😬

> Should people be encouraged to install envisalink_new via HACS from this repo?

I won't discourage anyone.  I do want to make sure I'm not creating conflict with HA - although I'm not sure if they'd care.

I thought about getting it added to HACS as a default repository but my research into that lead to [here](https://hacs.xyz/docs/publish/include) which I think indicates it would not be accepted due to the "override" clause.

> Will you publish your updated pyenvisalink to pypi?

I already did as it was required to submit the PR to HA.  It's just a [beta](https://pypi.org/project/pyenvisalink/5.0.0b1/) release for now though so doesn't show up using a default search.

> Do you want contributions for updated README's for these two repos?

Always happy to accept help!

In case anyone is interested, my branch of HA with these changes integrated to replace the stock envisalink integration is [here](https://github.com/ufodone/home-assistant-core/tree/envisalink_config_flow).

**Comment by vednolacni** [comment]
@ufodone - Thank you for your work. 

As a pire consumer im wiling to contribute financially, if that will help you make time for this. Do you have buy me a coffee or something similar?

**Comment by rct** [comment]
@ufodone - From my perspective, the important thing here is to make your hard work as easily accessible to envisalink users since it fixes a number of important bugs.

For now the two most common routes, getting HA core to accept the PR and becoming a HACS default repo, are unavailable.  So that leaves getting the word out to envisalink users how to add the custom repo and install envisalink new from there. 

If that sounds right, I’ll submit updated READMEs for pyenvisalink and envisalink_new that will help fill in the details, ok?

**Comment by ufodone** [comment]
@vednolacni Thanks for the offer but that's not necessary.  Available time is my blocker here.

@rct Thanks for this!

I've got a few other project around the house that have been languishing because of this work and I need to get back to them.  But I'll try and pick away at getting this work split up and submitted piecemeal.  It's likely to be a fairly lengthy process so I'll try and chip away at it a bit at a time.

**Comment by danTHAman152000** [comment]
I am a new Honeywell Vista 20p + Envisalink DUO user and added them to my HA nicely. I also tried the Homebridge Envisalink plugin. It had a feature that added a "bypass switch" that was convenient ... hit the switch if a door is open and the alarm was able to then bypass any door that was open and arm the alarm normally. I don't see a similar feature for the Envisalink_new plugin. My use case is for HomeKit and also wall-mounted tablets (which is why I require HA and not just HB). Is this feature possible to add?

**Comment by tinuva** [comment]
Its there for DSC panels. Unsure if for honeywell.

Go devices -> envisalink_new -> configure -> advanced 
tap "create bypass"

![Screenshot 2023-03-13 at 10 47 12](https://user-images.githubusercontent.com/5417682/224651282-7595ba20-ec1c-4dcc-ae6e-d9b6eb0f3ee2.png)

![Screenshot 2023-03-13 at 10 47 18](https://user-images.githubusercontent.com/5417682/224651301-a1df2ea5-bf08-4326-a12e-a78cd9927455.png)

![Screenshot 2023-03-13 at 10 47 24](https://user-images.githubusercontent.com/5417682/224651305-930a2742-c405-4e44-82f6-381dda6d14a0.png)

**Comment by danTHAman152000** [comment]
Unfortunately it’s not there for me. Since this feature is available with Homebridge, I’d imagine it can the Honeywell can work with this plugin too. Hopefully others want this feature implemented as well.

Thanks for the quick response.

**Comment by jl-678** [comment]
@danTHAman152000 I suggest that you create a new issue for this feature. It will help the author track this.

**Comment by hunstad** [comment]
``
``Great work!
---
Background:
- Vista 21 with Envisalink 4.  Firmware 01.04.202A
- I have partitions 1&2, zones 2-8, 10-18.  
- A mixture of doors (5), windows (7) and motion sensors(4).  All sensors are wired.
 
----
**Testing results:**
---
**Testing sequence one:**
- shutdown my production HA
- stopped VM for my production HA
----
- Creation of new VM for Home Assistant on my Proxmox server
- Fresh install of latest version of Home Assistant
- Installed Glances, Log Viewer and SSH
- Restarted
- Install HACS
- Restarted
- Installed Envisalink_new (0.6.4)
- Restarted
- Configured Envisalink new
- I have partitions 1&2, zones 2-8, 10-18 but I configured it as zones 1-19
- Restarted HA
- Confirmed zones 1-19 configured with partitions 1&2
- Renamed several zones
- Reconfigured to what I really have (partitions 1&2, zones 2-8, 9-18)
- Confirmed ability to delete entities for zones 1, 10 and 19 
- Opened /closed all zones one by one and confirmed that the entity for the zone showed the correct state as well as the message for the keypad was correct.
- shutdown testing HA
- shutdown testing VM
----------------------------------------
**Testing sequence two:**
- start VM for my production HA
- start for my production HA
- added envisalink_new
- restarted HA
- it did not pick up my existing configuration
I did get a message:  "Retrying setup: Unrecognized/undetermined panel type at nnn.nnn.n.12:4025"  (I masked the IP)
- manually configured envisalink_new
- removed include statement for envisalink in my configuration.yaml
- restarted HA
- confirmed that all partitions and zones were  created with the default names
- manually renamed all zones and partitions
- Opened /closed all zones one by one and confirmed that the entity for the zone showed the correct state as well as the message for the keypad was correct.

----
- it's a nice day here in Colorado so I opened most of the windows to air out my house.
- Screen shot of the history showing the state status of what I had opened is below:
- Observations
- I was not closing/opening the windows during this timeframe.  The history was showing I was closing the windows with increasing duration for about 12 cycles then restarting.  
- It missed several basement motion events
- it missed several LR motion events.  It failed to pick up a LR motion clear event that occured several seconds after the detection.
- 
<img width="1491" alt="Screen Shot 2023-03-14 at 3 04 24 PM" src="https://user-images.githubusercontent.com/13173018/225140603-35b7c10d-12b6-4214-8931-474cf08ee776.png">

Detail of  just one of the window entities showing curious progression of invalid "closed" durations: 
<img width="1409" alt="Screen Shot 2023-03-14 at 2 57 58 PM" src="https://user-images.githubusercontent.com/13173018/225143971-54b08e3d-a079-4bc5-b10c-a4578811fbcc.png">




- Like the original Envisalink integration, my logfile is flooded with state changes from the keypad:

<img width="472" alt="Screen Shot 2023-03-14 at 3 34 11 PM" src="https://user-images.githubusercontent.com/13173018/225144376-d1ac9d52-3923-4fbc-8bb5-894a69346fc2.png">



---

Clicking on the documentation link in the HA integration page takes you to: "https://www.home-assistant.io/integrations/envisalink_new" and a "Oh no! This page does not exist 😞"

---
- Logging set at debug
- Errors in the logfile:

```
2023-03-14 15:41:27.584 ERROR (MainThread) [pyenvisalink.envisalink_base_client] The server closed the connection. Reconnecting...
2023-03-14 15:41:57.588 ERROR (MainThread) [pyenvisalink.envisalink_base_client] The server closed the connection. Reconnecting...
2023-03-14 15:42:27.591 ERROR (MainThread) [pyenvisalink.envisalink_base_client] The server closed the connection. Reconnecting...
2023-03-14 15:42:57.595 ERROR (MainThread) [pyenvisalink.envisalink_base_client] The server closed the connection. Reconnecting...
2023-03-14 15:43:27.598 ERROR (MainThread) [pyenvisalink.envisalink_base_client] The server closed the connection. `Reconnecting...```
```


Let me know if you need more details.

**Comment by progamer562** [comment]
#31

**Comment by progamer562** [comment]
#31 well

**Comment by najikarim** [comment]
Thank you for your efforts...
1. Running v06.6
2. Dsc panel- I believe it's evl-3
3. I'm seeing the status of all of my 15 zones.
4. Integration is creating 17 entities
5. I'm not seeing the option under Advanced Config to create the Bypass switches nor am I seeing in devices/entities.

I had created a lot scripts that need to be modified to mimic the control panel using the Envisalink integration. Once I can sort the Bypass issue, which is my primary requirement, then I will update the automation/scripts/etc....

**Comment by jhenkens** [comment]
Very happy to have found this repo! I had a branch last fall that was trying to introduce `availability` and config flow to `envisalink`, but never finished it. Was about to pick it up again, and found this.

Tested a new config-flow entry, as well as migrating existing entry (and removing from YAML after migration). Only hiccup was renaming my script from `envisalink.send_keypress` to `envisalink_new.send_keypress`. Thankfully I have a helper script that prepends the alarm-code from my secrets file, so I only needed to change it in one place.

```
Envisalink 4: HONEYWELL
by eyezon
Firmware: 01.05.203A
Hardware: 4
```

I did get one transient error during reboots
```Logger: custom_components.envisalink_new.pyenvisalink.alarm_panel
Source: custom_components/envisalink_new/pyenvisalink/alarm_panel.py:461
Integration: Envisalink_new
First occurred: 12:53:25 PM (1 occurrences)
Last logged: 12:53:25 PM

Unable to validate connection: TimeoutError()```
